pkg/compactor/compactor_test.go

```diff
@@ -190,7 +190,8 @@ func TestMultitenantCompactor_ShouldDoNothingOnNoUserBlocks(t *testing.T) {
 
 		# TYPE cortex_compactor_runs_failed_total counter
 		# HELP cortex_compactor_runs_failed_total Total number of compaction runs failed.
-		cortex_compactor_runs_failed_total 0
+		cortex_compactor_runs_failed_total{reason="error"} 0
```

Comment:
Can we have a unit test where we call `compactUsers()` with a canceled context to ensure the right metric is tracked?

Reply:
that was a good call. It doesn't work because we wrap the `context.Cancelled` in a `multierr.MultiError`, which doesn't implement `errors.Unwrap` 😬

Reply:
i opened https://github.com/grafana/dskit/pull/254 in dskit to implement `Is(error) bool`

Reply:
the dskit change was merged and i revendored dskit in this PR. The other changes to dskit were minor, so I thought it's ok to make the vendoring and the compactor metric in the same PR